File: app/src/main/java/org/lineageos/jelly/utils/FileUtils.java
package org.lineageos.jelly.utils;

import android.support.annotation.NonNull;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

public class FileUtils {
    @NonNull
    public static String readStringFromStream(@NonNull InputStream inputStream,
                                              @NonNull String encoding) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream, encoding));
        StringBuilder result = new StringBuilder();
        String line;
        while ((line = reader.readLine()) != null) {
            result.append(line);
        }
        return result.toString();
    }
}

File: app/src/main/java/org/lineageos/jelly/utils/FileUtils.kt
/*
 * SPDX-FileCopyrightText: 2020 The LineageOS Project
 * SPDX-License-Identifier: Apache-2.0
 */

package org.lineageos.jelly.utils

import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader

object FileUtils {
    fun readStringFromStream(inputStream: InputStream,
                             encoding: String): String {
        val reader = BufferedReader(InputStreamReader(inputStream, encoding))
        val result = StringBuilder()
        var line: String?
        while (reader.readLine().also { line = it } != null) {
            result.append(line)
        }
        return result.toString()
    }
}
